server/src/main/java/org/apache/iotdb/db/query/fill/PreviousFill.java

```diff
@@ -59,28 +84,184 @@ public long getBeforeRange() {
+    TimeValuePair lastPoint = new TimeValuePair(0, null);
+    for (int index = seqFileResource.size() - 1; index >= 0; index--) {
+      TsFileResource resource = seqFileResource.get(index);
+      TimeseriesMetadata timeseriesMetadata =
+          FileLoaderUtils.loadTimeSeriesMetadata(
+              resource, seriesPath, context, timeFilter, allSensors);
+      if (timeseriesMetadata != null) {
+        if (endtimeContainedByTimeFilter(timeseriesMetadata.getStatistics())) {
+          return constructLastPair(
+              timeseriesMetadata.getStatistics().getEndTime(),
+              timeseriesMetadata.getStatistics().getLastValue(),
+              dataType);
+        } else {
+          List<ChunkMetadata> seqChunkMetadataList =
+              FileLoaderUtils.loadChunkMetadataList(timeseriesMetadata);
+
+          for (int i = seqChunkMetadataList.size() - 1; i >= 0; i--) {
+            lastPoint = getChunkLastPoint(seqChunkMetadataList.get(i));
+            // last point of this sequence chunk is valid, quit the loop
+            if (lastPoint.getValue() != null) {
+              return lastPoint;
+            }
+          }
+        }
+      }
+    }
+
+    return lastPoint;
+  }
+
+  /**
+   * find the last TimeseriesMetadata and unpack all overlapped seq/unseq files
+   */
+  private void UnpackOverlappedUnseqFiles(long lBoundTime) throws IOException {
+    PriorityQueue<TsFileResource> unseqFileResource =
+        sortUnSeqFileResourcesInDecendingOrder(dataSource.getUnseqResources());
+
+    while (!unseqFileResource.isEmpty()) {
+      // The very end time of unseq files is smaller than lBoundTime,
+      // then skip all the rest unseq files
+      if (unseqFileResource.peek().getEndTimeMap().get(seriesPath.getDevice()) < lBoundTime) {
+        return;
+      }
+      TimeseriesMetadata timeseriesMetadata =
+          FileLoaderUtils.loadTimeSeriesMetadata(
+              unseqFileResource.poll(), seriesPath, context, timeFilter, allSensors);
+      if (timeseriesMetadata != null
+          && lBoundTime <= timeseriesMetadata.getStatistics().getEndTime()) {
+        lBoundTime = Math.max(lBoundTime, timeseriesMetadata.getStatistics().getStartTime());
+        unseqTimeseriesMetadataList.add(timeseriesMetadata);
+        break;
+      }
+    }
+
+    // unpack all overlapped unseq files and fill unseqTimeseriesMetadata list
+    while (!unseqFileResource.isEmpty()
+        && (lBoundTime <= unseqFileResource.peek().getEndTimeMap().get(seriesPath.getDevice()))) {
+      TimeseriesMetadata timeseriesMetadata =
+          FileLoaderUtils.loadTimeSeriesMetadata(
+              unseqFileResource.poll(), seriesPath, context, timeFilter, allSensors);
+      unseqTimeseriesMetadataList.add(timeseriesMetadata);
+      // current unseq timeseriesMetadata's last point is a valid result,
+      // then skip the rest unseq files
+      if (endtimeContainedByTimeFilter(timeseriesMetadata.getStatistics())) {
```

Comment:
this maybe a bug. 

suppose the queried range is 1-100.

unseq files are [1,100]-version10, [10,100]-version5.
You find [10,100]'s endtime is contained and break. Then the [1,100] with higher priority is skipped.